napari/_vispy/layers/labels.py

```diff
@@ -120,60 +161,220 @@ def __init__(
         self,
         use_selection=False,
         selection=0.0,
+        collision=True,
+        color_count=1,
     ):
         colors = ['w', 'w']  # dummy values, since we use our own machinery
         super().__init__(colors, controls=None, interpolation='zero')
-        self.glsl_map = direct_lookup_shader.replace(
-            '$use_selection', str(use_selection).lower()
-        ).replace('$selection', str(selection))
+        self.glsl_map = (
+            direct_lookup_shader.replace(
+                "$use_selection", str(use_selection).lower()
+            )
+            .replace("$selection", str(selection))
+            .replace("$collision", str(collision).lower())
+            .replace("$color_count", str(color_count))
+        )
 
 
-def idx_to_2D(idx, shape):
+def idx_to_2d(idx, shape):
     """
     From a 1D index generate a 2D index that fits the given shape.
 
     The 2D index will wrap around line by line and back to the beginning.
     """
-    return (idx // shape[1]) % shape[0], (idx % shape[1])
+    return int((idx // shape[1]) % shape[0]), int(idx % shape[1])
 
 
-def hash2d_get(key, keys, values, empty_val=0):
+def hash2d_get(key, keys, col_key) -> Tuple[Optional[Tuple[int, int]], bool]:
```

Comment:
I thought it was like `column_key` at first

```suggestion
def hash2d_get(key, keys, collision_keys) -> Tuple[Optional[Tuple[int, int]], bool]:
```